### PR TITLE
add few more parameters #17

### PR DESCRIPTION
**Pull request by ar3ndt**

### CODE REVIEW

File: resources/terraform/modules/nodes/main.tf [hunk 1]
```diff
@@ -5,6 +5,12 @@ resource "aws_eks_node_group" "eks_nodes" {
   node_role_arn   = aws_iam_role.eks_nodes_iam_role.arn
   subnet_ids      = var.subnet_ids
   instance_types  = [var.worker_groups[count.index].instance_type]
+  disk_size       = var.disk_size
+  ami_type        = var.ami_type
+
+  remote_access {
+    ec2_ssh_key  = var.ec2_ssh_key
```
Comment (sk4zuzu) [inline resources/terraform/modules/nodes/main.tf:12]: Doesn't this force users to provide the key while it's not strictly necessary? :thinking:

Comment (ar3ndt) [reply resources/terraform/modules/nodes/main.tf:12]: It doesn't. First I wanted to use count with true/false condition checking if it is set but it applies only for 'resource'. Simply leaving null wont take it into account.

**Review by sk4zuzu — DISMISSED** [review]
LGTM

**Comment by sk4zuzu** [comment]
> LGTM

I think the issue states that we need more parameters:
- worker_groups [instance_type, asg_desired_capacity, asg_min_size, asg_max_size]

What happens with those, do we implement them in another PR? :thinking:

**Comment by ar3ndt** [comment]
> > LGTM
> 
> I think the issue states that we need more parameters:
> 
> * worker_groups [instance_type, asg_desired_capacity, asg_min_size, asg_max_size]
> 
> What happens with those, do we implement them in another PR? 🤔

True, these parameters are already introdued with defaults but not changeable by user. I will add this

**Comment by ar3ndt** [comment]
> > > LGTM
> > 
> > 
> > I think the issue states that we need more parameters:
> > 
> > * worker_groups [instance_type, asg_desired_capacity, asg_min_size, asg_max_size]
> > 
> > What happens with those, do we implement them in another PR? thinking
> 
> True, these parameters are already introdued with defaults but not changeable by user. I will add this

updated

File: resources/defaults.mk [hunk 2]
```diff
@@ -3,6 +3,15 @@ M_VPC_ID ?= unset
 M_SUBNET_IDS ?= null
 M_REGION ?= eu-central-1
 M_PRIVATE_ROUTE_TABLE_ID ?= unset
+M_DISK_SIZE ?= 36
```
Comment (atsikham) [inline resources/defaults.mk:6]: Why 36?

File: resources/defaults.mk [hunk 3]
```diff
@@ -3,6 +3,15 @@ M_VPC_ID ?= unset
...
+M_AUTOSCALER_SCALE_DOWN_UTILIZATION_THRESHOLD ?= 0.65
+M_EC2_SSH_KEY ?= null
+M_AMI_TYPE ?= AL2_x86_64
+M_WORKER_GROUPS_NAME ?= default_wg
+M_WORKER_GROUPS_INSTANCE_TYPE ?= t2.small
+M_WORKER_GROUPS_ADG_DESIRED_CAPACITY ?= 1
+M_WORKER_GROUPS_ASG_MIN_SIZE ?= 1
+M_WORKER_GROUPS_ASG_MAX_SIZE ?= 1
```
Comment (atsikham) [inline resources/defaults.mk:14]: What do you think to combine it in a [similar](https://github.com/epiphany-platform/m-azure-kubernetes-service/blob/develop/resources/defaults.mk#L12) way across modules? [This](https://github.com/epiphany-platform/m-azure-kubernetes-service/pull/28/commits/50304ecb12f2c0ec1d4e57a1f99cd0eb5c7f06e0) change is necessary to do that.

File: resources/terraform/main.tf [hunk 4]
```diff
@@ -16,20 +16,25 @@ module "nodes" {
...
-  autoscaler_chart_version = "7.3.4"
-  depends_on               = [module.control_plane, module.nodes]
+  source                                      = "./modules/autoscaler"
+  name                                        = var.name
+  region                                      = var.region
+  openid_connect_arn                          = module.control_plane.openid_connect_arn
+  openid_connect_url                          = module.control_plane.openid_connect_url
+  autoscaler_version                          = local.autoscaler_version
+  autoscaler_chart_version                    = "7.3.4"
+  depends_on                                  = [module.control_plane, module.nodes]
```
Comment (atsikham) [inline resources/terraform/main.tf:36]: please move this line to the bottom.

File: resources/terraform/main.tf [hunk 5]
```diff
@@ -16,20 +16,25 @@ module "nodes" {
...
-  openid_connect_url       = module.control_plane.openid_connect_url
-  autoscaler_version       = local.autoscaler_version
-  autoscaler_chart_version = "7.3.4"
-  depends_on               = [module.control_plane, module.nodes]
+  source                                      = "./modules/autoscaler"
+  name                                        = var.name
+  region                                      = var.region
+  openid_connect_arn                          = module.control_plane.openid_connect_arn
+  openid_connect_url                          = module.control_plane.openid_connect_url
+  autoscaler_version                          = local.autoscaler_version
```
Comment (atsikham) [inline resources/terraform/main.tf:34]: We don't need local values that are the same as variables.

File: resources/terraform/modules/autoscaler/locals.tf [hunk 6]
```diff
@@ -3,7 +3,7 @@ locals {
   k8s_service_account_name                    = "cluster-autoscaler-aws-cluster-autoscaler"
 
   # https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-does-scale-down-work
-  autoscaler_scale_down_utilization_threshold = "0.65"
+  autoscaler_scale_down_utilization_threshold = var.autoscaler_scale_down_utilization_threshold
```
Comment (atsikham) [inline resources/terraform/modules/autoscaler/locals.tf:6]: We don't need local values that are the same as variables.

**Review by atsikham — CHANGES_REQUESTED** [review]

File: resources/defaults.mk (hunk 2 above)
Comment (ar3ndt) [reply resources/defaults.mk:6]: the same as for azure

Comment (atsikham) [reply resources/defaults.mk:6]: But there is 32 - https://github.com/epiphany-platform/m-azure-basic-infrastructure/blob/887dd68a1f57ae5b22a41564ebbc45ad426fe43c/resources/terraform/modules/vms/main.tf#L78
If there is 36 somewhere, was that done by purpose?

File: resources/templates.mk [hunk 7]
```diff
@@ -15,6 +15,11 @@ $(M_MODULE_SHORT):
   region: $(M_REGION)
   subnet_ids: $(M_SUBNET_IDS)
   private_route_table_id: $(M_PRIVATE_ROUTE_TABLE_ID)
+  disk_size: $(M_DISK_SIZE)
+  autoscaler_scale_down_utilization_threshold: $(M_AUTOSCALER_SCALE_DOWN_UTILIZATION_THRESHOLD)
+  ami_type: $(M_AMI_TYPE)
+  ec2_ssh_key: $(M_EC2_SSH_KEY)
+  worker_groups: [$(M_WORKER_GROUPS)]
```
Comment (atsikham) [inline resources/templates.mk:22]: M_WORKER_GROUPS should be a list of objects. Did you test how specifying a few worker groups works?

File: resources/terraform/modules/nodes/main.tf (hunk 1 above)
Comment (atsikham) [reply resources/terraform/modules/nodes/main.tf:12]: @ar3ndt If you need that behaviour that you mentioned, there is [dynamic blocks](https://www.terraform.io/docs/configuration/expressions/dynamic-blocks.html) for that.
Now nothing should bechanged here.

**Review by atsikham — CHANGES_REQUESTED** [review]

**Comment by atsikham** [comment]
Please discuss with @seriva which pr - #29 or #30 will be merged first. I'm not 100% sure, but they may affect each other.

File: resources/defaults.mk (hunk 2 above)
Comment (ar3ndt) [reply resources/defaults.mk:6]: here it is 36.
https://github.com/epiphany-platform/m-azure-kubernetes-service/blob/develop/resources/defaults.mk

**Comment by ar3ndt** [comment]
> Please discuss with @seriva which pr - #29 or #30 will be merged first. I'm not 100% sure, but they may affect each other.

I this #29 should go first as it blocks pipeline

File: resources/defaults.mk (hunk 2 above)
Comment (atsikham) [reply resources/defaults.mk:6]: I suggest to leave default value as 32 as it's something commonfor disk size.

**Review by atsikham — APPROVED** [review]

**Comment by mkyc** [comment]
@ar3ndt please connect this PR to issue in ZH.

**Review by sk4zuzu — APPROVED** [review]
LGTM

**Pull request merged**